extensions/amp-viewer-integration/0.1/findtext.js

```diff
@@ -118,65 +215,76 @@ export function canonicalizeString(s) {
  * @return {?Array<!TextRangeDef>}
  */
 export function findSentences(win, node, sentences) {
+  sentences = canonicalizeSentences(sentences);
+  if (sentences.length <= 0) {
+    return null;
+  }
   const scanner = new TextScanner(win, node);
-  const matches = [];
-  for (let senIdx = 0; senIdx < sentences.length; senIdx++) {
-    const sen = canonicalizeString(sentences[senIdx]);
-    if (!sen) {
+  // Creates a circular buffer with capacity = max(size of sentence).
+  /** @type {!CircularBuffer<!TextPosAndIdxDef>} */
+  const buf = new CircularBuffer(
+    sentences.map(sen => sen.length).reduce((x, y) => Math.max(x, y))
+  );
+
+  // First, create a matcher for the first sentence to find only the first
+  // sentence.
+  /** @type {!Array<!Matcher>} */
+  const matchers = [new Matcher(sentences[0], buf)];
+  let posIdx = -1;
+  while (true) {
+    posIdx++;
+    const pos = scanner.next();
+    if (pos == null) {
```

Comment:
Should this be triple comparator `===`?

Reply:
`const pos` can not be `undefined`. So, I can make it `===`.
But do we really use `===` so that this condition becomes false when pos is `undefined`?
https://google.github.io/styleguide/jsguide.html#features-equality-checks-exceptions

Reply:
Cool, thanks for clarifying.